### ISSUE TITLE
Compile error on a method with @JSName("finalize") in a non-native JS class

### ISSUE THREAD

**Issue by unarist** [scala-js/scala-js#3939]
```scala
// based on https://github.com/scala-js/scalajs-tutorial/blob/0.6.x/src/main/scala/tutorial/webapp/TutorialApp.scala
package tutorial.webapp

import scala.scalajs.js

@js.annotation.ScalaJSDefined
class Test extends js.Object {
  @js.annotation.JSName("finalize")
  def foo(): Unit = {
    println("success")
  }
}

object TutorialApp {
  def main(args: Array[String]): Unit = {
    new Test().foo()
    println("hoge")
  }
}
```

This code should print "success" and "hoge", but it cannot be compiled:

Scala 2.12.10 / ScalaJS 0.6.31 (and 1.0.0-RC2)

```
sbt:Scala.js Tutorial> run
[info] Compiling 1 Scala source to D:\hoge\target\scala-2.12\classes ...
[warn] D:\hoge\src\main\scala\tutorial\webapp\TutorialApp.scala:5:2: @ScalaJSDefined is deprecated: add `-P:scalajs:sjsDefinedByDefault` to your scalac options and simply remove `@ScalaJSDefined`
[warn] @js.annotation.ScalaJSDefined
[warn]  ^
[error] D:\hoge\src\main\scala\tutorial\webapp\TutorialApp.scala:8:7: Cannot disambiguate overloads for method finalize with types
[error]   ()Unit
[error]   ()Unit
[error]   def foo(): Unit = {
[error]       ^
[warn] one warning found
[error] one error found
[error] (Compile / compileIncremental) Compilation failed
[error] Total time: 1 s, completed 2020/01/22 22:03:20
```

In ScalaFiddle, it can be compiled but it only prints "hoge": https://scalafiddle.io/sf/rPmTeX6/0

**Comment by sjrd** [comment]
Thanks for the report.

This happens because the method then clashes with `def finalize(): Unit` declared in `java.lang.Object`. Although all JS objects statically extend `j.l.Object`, its methods, like `finalize`, are not actually exposed as public JS methods. They have a fixed, non-overridable behavior for JS objects. So logically they should not clash. We might have to teach the JSName override checker that this is fine.

It's actually also possible to simply declare an `override def finalize(): Unit`, which I believe will pass the check, *and even do the "right" thing*. "Right" in quotes because that method will be exposed to JS, but it won't actually *override* `Object.finalize`.

I assume the same issue happens with other methods defined in `j.l.Object`, e.g., `hashCode`, `wait`, etc. `toString()` is special and handled correctly, of course.

**Comment by sjrd** [comment]
Fixed in 902f151457f02d2b7e650e7545381303c85681cd.
